packages/customer-account-ui-extensions/src/extension-points/standard-api/navigation.ts

```diff
@@ -0,0 +1,30 @@
+export type NavigationType = 'push' | 'replace' | 'traverse';
+
+export interface NavigationOptions {
+  state?: Record<string, any>;
+  type: 'push' | 'replace';
+}
+
+export interface NavigationHistoryEntry {
+  key: string;
+  url: string;
+  state: Record<string, any>;
+}
+
+export interface NavigationCurrentEntryChangeEvent {
+  type: NavigationType;
+  from: NavigationHistoryEntry;
+}
+
+export interface StandardExtensionNavigation {
+  navigate: NavigateFunction;
+}
+
+export type FullExtensionNavigation = StandardExtensionNavigation & {
+  currentEntry: NavigationHistoryEntry;
```

Comment:
Will add the listeners in a follow up pr when Implemented in customer-account-web